### PR TITLE
Hyperlink fix

### PR DESCRIPTION
**Pull request by feipenghe**
This PR closes  https://github.com/asyml/forte/issues/669 https://github.com/asyml/forte/issues/667 and partially corrects https://github.com/asyml/forte/issues/610

### Description of changes
* Fix invalid hyperlinks.
* add classes under `ft` and `ftx` folders into rst files and it shows on main documentation page. 

### Possible influences of this PR.
Describe what are the possible side-effects of the code change.

### Test Conducted
Describe what test cases are included for the PR.

Progress tracking:
- [x] Common
   - [x] Exceptions
   - [x] Resources
- [x] Data
   - [x] Converter
   - [x] Extractors
   - [x] Ontology
   - [x] Readers
   - [x] Other data modules
- [x] Datasets
- [x] Evaluation
- [x] models
- [x] processors
- [x] pipeline

ontology classes documentation effects
image

### CODE REVIEW

File: docs/spelling_wordlist.txt
Comment (hunterhector) [inline docs/spelling_wordlist.txt:170]: Most of these words here are variables/programmatic terms, they are not supposed to be in the wordlist. You should wrap those with backtip, like this: `str`

File: docs/code/ft.rst
Comment (hunterhector) [inline docs/code/ft.rst:5]: I am wondering whether we need to document the api for all generated ontology classes.

Comment (hunterhector) [reply docs/code/ft.rst:5]: yes, I mean I am considering whether we should include these. Maybe it is a bit much to write all the classes in documentation from the generated classes. But I am not sure of the answer yet.

File: forte/processors/data_augment/algorithms/text_replacement_op.py
Comment (hunterhector) [inline forte/processors/data_augment/algorithms/text_replacement_op.py:50]: why is this changed to a comma?

Comment (feipenghe) [reply forte/processors/data_augment/algorithms/text_replacement_op.py]: Thought I could fix another bug. I have changed it back.

**Review by hunterhector — COMMENTED** [review]
I found a few indent problems, let me just list them below. Please double-check similar problems in other places.

- https://asyml-forte--668.org.readthedocs.build/en/668/code/data.html#forte.data.data_pack.DataIndex. `Returns: Entry ID of type inner_type that is covered by`
- Also on the same page, sometimes `Returns` is bolded, sometimes it is not. Maybe we should make it consistent.
- https://asyml-forte--668.org.readthedocs.build/en/668/code/data.html#forte.data.selector.NameMatchSelector, sometimes "Deprecated" is shaded, sometimes it is not.
- https://asyml-forte--668.org.readthedocs.build/en/668/code/data.html#forte.data.data_pack_dataset.DataPackIterator, this line `Here is an example usage` probably should not be shaded.
- https://asyml-forte--668.org.readthedocs.build/en/668/code/data.html#forte.data.batchers.FixedSizeDataPackBatcher.default_configs, maybe `Here` should not be shaded
- https://asyml-forte--668.org.readthedocs.build/en/668/code/data.html#forte.data.entry_type_generator.EntryTypeGenerator.get_entry_attributes_by_class, `note` and `code block` not working.
- https://asyml-forte--668.org.readthedocs.build/en/668/code/pipeline.html#forte.process_manager.ProcessManager.Args, format here is wrong
- https://asyml-forte--668.org.readthedocs.build/en/668/code/processors.html#forte.processors.misc.lowercaser_processor.LowerCaserProcessor.default_configs, format here is wrong
- https://asyml-forte--668.org.readthedocs.build/en/668/code/data_aug.html#forte.processors.data_augment.algorithms.back_translation_op.BackTranslationOp, format here looks wrong
- https://asyml-forte--668.org.readthedocs.build/en/668/code/data_aug.html#forte.processors.data_augment.algorithms.dictionary_replacement_op.DictionaryReplacementOp, `The config should contain the following fields` might not be shaded
- https://asyml-forte--668.org.readthedocs.build/en/668/code/data_aug.html#forte.processors.data_augment.algorithms.typo_replacement_op.TypoReplacementOp, format here looks weird
- https://asyml-forte--668.org.readthedocs.build/en/668/code/data_aug.html#forte.processors.data_augment.algorithms.word_splitting_processor.RandomWordSplitDataAugmentProcessor.default_configs, config here looks wrong
-

**Comment by feipenghe** [comment]
> * https://asyml-forte--668.org.readthedocs.build/en/668/code/data.html#forte.data.selector.NameMatchSelector

I am thinking maybe we can merge this one first and then create another PR for the styling issue. Since I have other branches depending on this branch, and they can be merged after this PR is merged.

**Comment by hunterhector** [comment]
> > * https://asyml-forte--668.org.readthedocs.build/en/668/code/data.html#forte.data.selector.NameMatchSelector
> 
> I am thinking maybe we can merge this one first and then create another PR for the styling issue. Since I have other branches depending on this branch, and they can be merged after this PR is merged.

Ok, please make sure these are tracked.

**Review by hunterhector — APPROVED** [review]

**Pull request merged**
